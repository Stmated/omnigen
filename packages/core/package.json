{
  "name": "@omnigen/core",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "scripts": {
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "dev": "tsc -w",
    "lint": "TIMING=1 eslint src/**/*.ts* --fix",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "dependencies": {
    "@omnigen/core-log": "workspace:*",
    "@etclabscore/string-json-template-language": "^1.0.0",
    "@open-rpc/schema-utils-js": "^1.16.1",
    "change-case": "^4.1.2",
    "deepmerge": "^4.3.0",
    "js-yaml": "^4.1.0",
    "json-pointer": "^0.6.2",
    "node-fetch": "^2.6.9",
    "object-hash": "^3.0.0",
    "string-similarity": "^4.0.4",
    "ts-json-schema-generator": "^1.2.0",
    "zod": "^3.22.4"
  },
  "devDependencies": {
    "@omnigen/config-typescript": "workspace:*",
    "@omnigen/eslint-config-custom": "workspace:*",
    "@swc/core": "^1.3.32",
    "@types/json-pointer": "^1.0.31",
    "@types/json-schema": "^7.0.15",
    "@types/node": "^18.13.0",
    "@types/node-fetch": "^2.6.2",
    "@types/object-hash": "^2.2.1",
    "@types/string-similarity": "^4.0.0",
    "colors": "^1.4.0",
    "nock": "^13.3.0",
    "typescript": "^5.4.2",
    "type-fest": "^3.6.1"
  }
}
