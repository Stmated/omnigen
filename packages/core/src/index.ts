import {PluginAutoRegistry} from '@omnigen/core-plugin';
import {CommonTransformPlugin, CommonTransform2Plugin, fileWriter, CorePlugin} from './CoreUtilPluginInit.ts';

export * from './ast';
export * from './equality';
export * from './parse';
export * from './util';
export * from './visit';
export * from './write';
export * from './CoreUtilPluginInit.ts';

export default PluginAutoRegistry.register([CorePlugin, CommonTransformPlugin, CommonTransform2Plugin, fileWriter]);
