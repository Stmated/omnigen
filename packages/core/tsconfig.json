{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "@omnigen/config-typescript/build.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "baseUrl": "./src"
  }
}
