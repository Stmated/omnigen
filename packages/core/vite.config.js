import {resolve} from 'path';
import {defineConfig} from 'vite';

export default defineConfig({
  build: {
    outDir: 'dist/lib',
    lib: {
      // Could also be a dictionary or array of multiple entry points
      entry: resolve(__dirname, 'src/index.ts'),
      name: '@omnigen/core',
      // the proper extensions will be added
      // fileName: 'core',
    },
    rollupOptions: {
      // make sure to externalize deps that shouldn't be bundled into your library
      external: ['node-fetch', 'promisify', 'node:fs', 'node:url', 'find-up', 'fs/promises', 'fs'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        // globals: {
        //   vue: 'Vue',
        // },
      },
      treeshake: 'safest',
    },
  },
  optimizeDeps: {
    exclude: ['sync-fetch'],
  },
});
