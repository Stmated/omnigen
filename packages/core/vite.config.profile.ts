import {resolve} from 'path';
import {type UserConfig} from 'vite';

export default {
  build: {
    outDir: 'dist/profile',
    minify: false,
    terserOptions: {
      compress: false,
      mangle: false,
    },
    lib: {
      // Could also be a dictionary or array of multiple entry points
      entry: resolve(__dirname, 'src/index.profiling.ts'),
      name: '@omnigen/core',
      // the proper extensions will be added
      // fileName: 'core',
    },
    rollupOptions: {
      external: ['node-fetch', 'promisify', 'node:fs', 'node:url', 'find-up'],
      output: {
        // Provide global variables to use in the UMD build for externalized deps
        // globals: {
        //   vue: 'Vue',
        // },
      },
    },
  },
} satisfies UserConfig;
