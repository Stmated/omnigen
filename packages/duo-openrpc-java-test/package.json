{
  "name": "@omnigen/duo-openrpc-java-test",
  "version": "0.0.0",
  "private": true,
  "license": "MIT",
  "type": "module",
  "main": "./src/index.ts",
  "types": "./src/index.ts",
  "scripts": {
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "dev": "tsc -w",
    "lint": "TIMING=1 eslint src/**/*.ts* --fix",
    "test": "NODE_OPTIONS=--experimental-vm-modules jest --passWithNoTests",
    "test:watch": "jest --watch --passWithNoTests"
  },
  "jest": {
    "preset": "@omnigen/config-jest"
  },
  "dependencies": {
    "@omnigen/core": "0.0.0",
    "@omnigen/core-log": "0.0.0",
    "@omnigen/core-util": "0.0.0",
    "@omnigen/parser-jsonschema": "0.0.0",
    "@omnigen/parser-openrpc": "0.0.0",
    "@omnigen/target-java": "0.0.0",
    "json-schema": "^0.4.0"
  },
  "devDependencies": {
    "@types/jest": "^29.4.0",
    "@types/json-schema": "^7.0.11",
    "@omnigen/config-jest": "0.0.0",
    "@omnigen/config-typescript": "0.0.0",
    "@omnigen/eslint-config-custom": "0.0.0",
    "@omnigen/utils-test": "0.0.0",
    "@omnigen/utils-test-target-java": "0.0.0",
    "typescript": "beta",
    "jest": "^29.4.1",
    "java-parser": "^2.0.3"
  }
}
