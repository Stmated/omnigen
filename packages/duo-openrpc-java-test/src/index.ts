
export * from './JavaTestUtils.js';
export * from './OpenRpcTestUtils.js';
