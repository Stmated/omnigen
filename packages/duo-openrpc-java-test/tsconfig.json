{
  "$schema": "https://json.schemastore.org/tsconfig",
  "extends": "@omnigen/tsconfig/build.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "baseUrl": "./src"
  }
}
