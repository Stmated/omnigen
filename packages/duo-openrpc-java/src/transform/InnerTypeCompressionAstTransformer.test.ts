import {DEFAULT_TEST_TARGET_OPTIONS, JavaTestUtils} from '@omnigen/duo-openrpc-java-test';
import {DEFAULT_MODEL_TRANSFORM_OPTIONS} from '@omnigen/core';
import {describe, expect, test, vi} from 'vitest';

describe('InnerTypeCompression', () => {

  test('CompressNo', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('multiple-inheritance.json', {
      modelTransformOptions: {...DEFAULT_MODEL_TRANSFORM_OPTIONS, generifyTypes: false, elevateProperties: false},
      targetOptions: {
        ...DEFAULT_TEST_TARGET_OPTIONS,
        compressSoloReferencedTypes: false,
        compressUnreferencedSubTypes: false,
      },
    });

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });

  test('CompressYes', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('multiple-inheritance.json', {
      modelTransformOptions: {...DEFAULT_MODEL_TRANSFORM_OPTIONS, generifyTypes: false, elevateProperties: false},
      targetOptions: {
        ...DEFAULT_TEST_TARGET_OPTIONS,
        compressSoloReferencedTypes: true,
        compressUnreferencedSubTypes: true,
      },
    });

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }

    // TODO: This is actually INCORRECT! It is not properly serializable with the @JsonValue annotation!
    //        Need to create some other way of handling this, like splitting into many different classes with unique parents

    // 'GiveInGetOut2Response.java', // TODO: WRONG! Should be created from give_string_get_string_result. Is closest "name"
  });

  test('CompressYes-error-structure', async () => {

    const fileContents = await JavaTestUtils.getFileContentsFromFile('error-structure.json', {
      modelTransformOptions: {...DEFAULT_MODEL_TRANSFORM_OPTIONS, generifyTypes: false, elevateProperties: false},
      targetOptions: {
        ...DEFAULT_TEST_TARGET_OPTIONS,
        compressSoloReferencedTypes: true,
        compressUnreferencedSubTypes: true,
      },
    });

    // NOTE: This *could* be more aggressive, but it starts to become confusing merging some of them.
    //        Could be a future option to, for example, put ListThingsRequest inside JsonRpcRequest (which it inherits from).
    expect([...fileContents.keys()].sort()).toMatchSnapshot();
  });

  test('CompressYes-error-structure+generics', async ({task}) => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('error-structure.json', {
      modelTransformOptions: {...DEFAULT_MODEL_TRANSFORM_OPTIONS, generifyTypes: true, elevateProperties: false},
      targetOptions: {
        ...DEFAULT_TEST_TARGET_OPTIONS,
        compressSoloReferencedTypes: true,
        compressUnreferencedSubTypes: true,
      },
    });

    expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${task.suite.name}/${task.name}/${fileName}`);
    }
  });
});
