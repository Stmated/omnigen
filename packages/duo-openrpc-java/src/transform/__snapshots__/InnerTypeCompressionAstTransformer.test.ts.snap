// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InnerTypeCompression > CompressNo 1`] = `
[
  "A.java",
  "AXOrB.java",
  "Abs.java",
  "B.java",
  "C.java",
  "ErrorUnknown.java",
  "ErrorUnknownError.java",
  "GiveInGetOut2Request.java",
  "GiveInGetOut2RequestParams.java",
  "GiveInGetOut2Response.java",
  "GiveInGetOutRequest.java",
  "GiveInGetOutRequestParams.java",
  "GiveInGetOutResponse.java",
  "IB.java",
  "IC.java",
  "In.java",
  "JsonRpcError.java",
  "JsonRpcErrorResponse.java",
  "JsonRpcRequest.java",
  "JsonRpcRequestParams.java",
  "JsonRpcResponse.java",
  "Out.java",
  "Out2.java",
]
`;

exports[`InnerTypeCompression > CompressYes 1`] = `
[
  "A.java",
  "AXOrB.java",
  "Abs.java",
  "C.java",
  "ErrorUnknown.java",
  "GiveInGetOut2Request.java",
  "GiveInGetOut2Response.java",
  "GiveInGetOutRequest.java",
  "GiveInGetOutResponse.java",
  "IB.java",
  "IC.java",
  "In.java",
  "JsonRpcError.java",
  "JsonRpcErrorResponse.java",
  "JsonRpcRequest.java",
  "JsonRpcRequestParams.java",
  "JsonRpcResponse.java",
]
`;

exports[`InnerTypeCompression > CompressYes-error-structure 1`] = `
[
  "ErrorUnknown.java",
  "JsonRpcError.java",
  "JsonRpcErrorResponse.java",
  "JsonRpcRequest.java",
  "JsonRpcRequestParams.java",
  "JsonRpcResponse.java",
  "ListThingsError100.java",
  "ListThingsRequest.java",
  "ListThingsResponse.java",
]
`;

exports[`InnerTypeCompression > CompressYes-error-structure+generics 1`] = `
[
  "ErrorUnknown.java",
  "JsonRpcError.java",
  "JsonRpcErrorResponse.java",
  "JsonRpcRequest.java",
  "JsonRpcRequestParams.java",
  "JsonRpcResponse.java",
  "ListThingsError100.java",
  "ListThingsRequest.java",
  "ListThingsResponse.java",
]
`;
